### PR TITLE
Generate Formula with AI 

### PR DESCRIPTION
**Pull request by mayneyao**
https://github.com/user-attachments/assets/3acedeb4-a60d-4c0d-a532-ddf5c7230f1f

### CODE REVIEW

File: apps/nextjs-app/src/lib/withEnv.ts [hunk 1]
```diff
@@ -15,12 +15,27 @@ type GetServerSideProps<
   D extends PreviewData = PreviewData,
 > = (context: GetServerSidePropsContext<Q, D>) => Promise<GetServerSidePropsResult<P>>;
 
+async function fetchPublicConfig(context?: GetServerSidePropsContext) {
+  try {
+    const protocol = process.env.NODE_ENV === 'development' ? 'http' : 'https';
```
Comment (tea-artist) [inline apps/nextjs-app/src/lib/withEnv.ts]: you can use axios from `apps/nextjs-app/src/backend/api/rest/axios.ts`
it design to handle internal request

File: apps/nextjs-app/src/lib/withEnv.ts [hunk 2]
```diff
@@ -31,6 +46,7 @@ export default function withEnv<P extends { [key: string]: any }>(
         sentryDsn: process.env.SENTRY_DSN,
         socialAuthProviders: process.env.SOCIAL_AUTH_PROVIDERS?.split(','),
         storagePrefix: process.env.STORAGE_PREFIX,
+        globalSettings: publicConfig,
```
Comment (tea-artist) [inline apps/nextjs-app/src/lib/withEnv.ts]: If it's just ssr, it doesn't need to be defined here.

**Review by tea-artist — DISMISSED** [review]

**Comment by tea-artist** [comment]
request progress 🥹

**Comment by mayneyao** [comment]
> request progress 🥹

soon🤓

**Comment by tea-artist** [comment]
Move to #1117

**Pull request closed without merging**